### PR TITLE
Code quality

### PR DESCRIPTION
**Pull request by cschroed-usgs**
Instrumenting this project via the maven code quality pom

### CODE REVIEW

**Comment by isuftin** [comment]
It does seem like the Clair scanner is having issues.

```
2019/10/28 19:14:33 [CRIT] ▶ Could not analyze layer: Clair responded with a failure: Got response 400 with message {"Error":{"Message":"could not find layer"}}
```

iirc this points to a misconfiguration with how the clair scanner is launched

File: pom.xml [hunk 1]
```diff
@@ -181,15 +205,6 @@
             <plugin>
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
```
Comment (isuftin) [inline pom.xml]: I wouldn't remover a version definition from any dependency. You will end up getting surprises down the line. Unless this is handled by a parent dependencymanagement in maven?

File: Dockerfile [hunk 2]
```diff
@@ -5,12 +5,29 @@ FROM maven@sha256:b37da91062d450f3c11c619187f0207bbb497fc89d265a46bbc6dc5f17c02a
...
+	fi
+
 COPY pom.xml /build/pom.xml
 WORKDIR /build
 
 #download all maven dependencies (this will only re-run if the pom has changed)
 RUN mvn -B dependency:go-offline
 
+# copy git history into build image so that sonar can report trends over time
+COPY .git /build
```
Comment (isuftin) [inline Dockerfile]: Eh. This grows the final image but only adds value to reporting around the project itself. Unless this image is used as a base image for a downstream build that pulls artifacts out of this image and discards everything else?

File: pom.xml (hunk 1 above)
Comment (cschroed-usgs) [reply pom.xml]: The parent pom defines it.

File: Dockerfile (hunk 2 above)
Comment (cschroed-usgs) [reply Dockerfile]: Yeah the downstream image only grabs the jar

**Comment by cschroed-usgs** [comment]
> It does seem like the Clair scanner is having issues.
> 
> ```
> 2019/10/28 19:14:33 [CRIT] ▶ Could not analyze layer: Clair responded with a failure: Got response 400 with message {"Error":{"Message":"could not find layer"}}
> ```
> 
> iirc this points to a misconfiguration with how the clair scanner is launched

Thanks for noticing this. I'm kinda surprised the clair scan would fail as a result of the proposed changes. I'll think more about that. In the mean time I'm re-running the Travis build in master to see if this is a time-variant thing.

https://travis-ci.com/USGS-CIDA/aqcu-vdi-report/jobs/189765226

**Comment by cschroed-usgs** [comment]
Thanks Ivan! Looks like switching from `eth0` to `docker0` fixed the clair scan. 
https://github.com/USGS-CIDA/aqcu-vdi-report/pull/14/commits/3cd535617a580c1de06aea69dae5c04e0f47f387

**Comment by isuftin** [comment]
Yikes that's a lot of CVEs. Thankfully none are medium/high

**Pull request merged**